src/partials/latestBlock/latestBlock.vue

```diff
@@ -69,26 +69,15 @@ export default {
   ]),
 
   /*
-   * mounted get the latest height of the
+   * get the latest height of the
    * blockchain, then get block data
    */
-  mounted: function () {
-    return this
-    .$store
-    .dispatch('blocks/height')
-    .then((blockHeight) => this
-    .$store
-    .dispatch('blocks/getBlockFromHeight', blockHeight))
-  },
-
   watch: {
-    height: function (val) {
-      return this
-        .$store
-        .dispatch('blocks/height')
-        .then((val) => this
-          .$store
-          .dispatch('blocks/getBlockFromHeight', val))
+    height: {
+      handler (val) {
+        this.$store.dispatch('blocks/getBlockFromHeight', val)
```

Comment:
It is better, but why you have removed `this.$store.dispatch('blocks/height')` call?

Reply:
I do not understand the purpose of that call. We are watching the height, changing it in the same function. If the watcher got called, that means that height just changed in state, doesn't that ensure that the height is latest?